examples/systemcalls/systemcalls.c

```diff
@@ -58,7 +124,24 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
-
+    
+    int status;
+    pid_t pid;
+    pid=fork();
+    if(pid==-1)
+    	return false;
+    else if(pid == 0)
+    {
+    	if((execv(full_path, rest_of_args))!=-1)
+    		return true;
+    	else
+    		return false;
```

Comment:
@abaa3023 I don't think returning here from the child process is going to do what you want.  I think you need an exit() in the failure case so you can send the failure message back to the parent.